torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1929,9 +2390,97 @@ def __init__(
         self.frozen = False
         self.dim_constraints: Optional[DimConstraints] = None
 
+        self.graph_added_nodes: Dict[Tuple[Callable, Tuple[Any, ...]], torch.fx.Node] = {}
+        self.source_to_symbol: Dict[str, sympy.Symbol] = {}
+        if _translation_validator_enabled():
+            self.validator = TranslationValidator()
+            self.graph = torch.fx.Graph()
+            # Create an output graph and start inserting before that.
+            # This is needed when 'deepcopy'-ing this object.
+            self.graph.inserting_before(self.graph.output(None))
+
     def freeze(self):
         self.frozen = True
 
+    def _create_symbol_for_source(self, source: Source, positive: Optional[bool]) -> Optional[sympy.Symbol]:
+        if _translation_validator_enabled():
+            srcname = source.name()
+            if source not in self.source_to_symbol:
+                self.source_to_symbol[srcname] = sympy.Symbol(srcname, positive=positive, integer=True)
```

Comment:
Are you sure you want to allocate a fresh sympy Symbol here??

Reply:
Well, that's the idea. This is so I make sure I have one unique symbol per source.